app.js

```diff
@@ -142,8 +142,8 @@ export async function configure(startStamp) {
         const request = require('request');
         const rewrite = require('express-urlrewrite');
         const proxy = require('http-proxy-middleware');
-        const uploadServer = `http://${hostname || 'localhost'}:${uport}`;
-        const downloadServer = `http://${hostname || 'localhost'}:${dport}`;
+        const uploadServer = `http://${hostname || 'uploader'}:${uport}`;
+        const downloadServer = `http://${hostname || 'downloader'}:${dport}`;
```

Comment:
Will we need to adjust 'hosts' file locally?

Reply:
As we run each service in separate container, we need to distinct uploader and downloader hosts.
As stated in documentation (https://docs.docker.com/compose/networking/)  "Each container for a service joins the default network and is both reachable by other containers on that network, and discoverable by them at a hostname identical to the container name." So, we need not modify anything when deploying as stack.

It will be a problem when running in traditional all-on-localhost environment, but this dirty hack is needed for containers to work.

The best solution, of course, will be to put this into config.